### ISSUE TITLE
Openrecoveryscript restore broken since version 2.8.7.0

### ISSUE THREAD

**Issue by andip71** [omnirom/android_bootable_recovery#38]
Hi,
I always used to create a script by an own app that just fires a single "restore xxxx" command in the script to restore a previous backup.

While this was running fine before TWRP 2.8.7.0, it now fails with the message:

  E: Cannot restore system -- mounted read only

I understand this is due to the latest changes on mounting the system volume ro for the OTA stuff.

However, it now completely breaks the restore command when placed into /cache/recovery/openrecoveryscript.

Is there any way I can workaround this issue? (I tried almost everything I can do in the script, but no dice)
Or can it be fixed in the next TWRP version again?
BTW... in an adb session when calling "twrp restore xxxx" it works fine too, just not when in the script itself.

Thanks
Andi

**Comment by xiaolu** [comment]
Edit twrp.cpp, move 335-352 line to 304.

**Comment by hene193** [comment]
Did you get this solved? I have this same problem on 3.0.0. I cannot restore with openrecoveryscript. Is there any workaround for this issue without recompiling the whole recovery?

**Comment by andip71** [comment]
Unfortunately this was not ever resolved.
I am not compiling recoveries myself, so I could not help myself.

So if you help pushing for a fix, much appreciated, @hene193.

**Comment by hene193** [comment]
There seems to be something really wrong in the code. As example the mount command does not work. I also tried cmd mount -o rw (and so on) and it did not also work. TWRP did not even mount it so that might be the problem. As for fixing it, I can't really help with that. It seems to be TWRP problem so I should maybe report this as a bug in the TWRP git...

**Comment by andip71** [comment]
Yep, I am happy to support you with my vote. Just let me know, it is really inconvenient like hell with that bug.

**Comment by yulinho** [comment]
have you guys solved the problem?

**Comment by andip71** [comment]
Unfortunately not so far :(

This bug seems to be completely ignored.

**Comment by yulinho** [comment]
I have to solve this problem because of I'm a second-hand cell phone supplier, I need to batch flash phone .I'm compiling the new 'recovery.img' now.
